src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/TabularOverloadsBlanksAndErrors.txt

```diff
@@ -0,0 +1,220 @@
+"Error(Kind=13), Error(Kind=13)"
+
+>> Concat(Find(Char(0), ["hello", "great", "world"]), IfError(Text(Result), $"Error(Kind={FirstError.Kind})"), ", ")
+"Error(Kind=25), Error(Kind=25), Error(Kind=25)"
+
+>> Concat(Find(["a", "b", "c"], "The quick brown fox jumped over the lazy dog", [1, 2, 3, 4, 5]), IfError(Text(Result), $"Error(Kind={FirstError.Kind})"), ", ")
+"38, 11, 8, Error(Kind=27), Error(Kind=27)"
+
+>> Round(If(1/0<2,[12.34, 56.78]), 1)
+#Error(Kind=Div0)
+
+>> Round(1234.5678, If(Sqrt(-1)<0,[1, 2, 3]))
+#Error(Kind=Numeric)
+
+>> Round([12.34, 56.78], 1/0)
+[Microsoft.PowerFx.Types.ErrorValue,Microsoft.PowerFx.Types.ErrorValue]
+
+>> Round(Sqrt(-1), [1, 2, 3])
+Table({Result:Microsoft.PowerFx.Types.ErrorValue},{Result:Microsoft.PowerFx.Types.ErrorValue},{Result:Microsoft.PowerFx.Types.ErrorValue})
+
+>> Round([123.456, 234.567], [1, 2, 3])
+[123.5,234.57,Microsoft.PowerFx.Types.ErrorValue]
+
+>> Concat(Round([12.34, 56.78], 1/0), IfError(Text(Value), $"Error(Kind={FirstError.Kind})"), ", ")
+"Error(Kind=13), Error(Kind=13)"
+
+>> Concat(Round(Sqrt(-1), [1, 2, 3]), IfError(Text(Result), $"Error(Kind={FirstError.Kind})"), ", ")
+"Error(Kind=24), Error(Kind=24), Error(Kind=24)"
+
+>> Concat(Round([123.456, 234.567], [1, 2, 3]), IfError(Text(Value), $"Error(Kind={FirstError.Kind})"), ", ")
+"123.5, 234.57, Error(Kind=27)"
+
+// *********** 2+-argument functions, blanks ***********
+
+>> Concatenate(If(1<0,["hello", "hi"]), " world")
+Blank()
+
+>> Concatenate("hello ", If(1<0,["world", "day"]))
+Blank()
+
+>> Concatenate(["hello", "hi"], If(1<0," world"))
+Table({Result:"hello"},{Result:"hi"})
+
+>> Concatenate(If(1<0,"hi"), ["hello", "great", "world"])
+Table({Result:"hello"},{Result:"great"},{Result:"world"})
+
+>> Find(If(1<0,["hello", "hi"]), " world")
+Blank()
+
+>> Find("a", If(1<0,["world", "day"]))
+Blank()
+
+>> Find(["a", "b"], If(1<0,"abcdefg"))
+Table({Result:Blank()},{Result:Blank()})
+
+>> Find(If(1<0,"a"), ["hello", "great", "world"])
+Table({Result:1},{Result:1},{Result:1})
+
+>> Round(If(1<0,[12.34, 56.78]), 1)
+Blank()
+
+>> Round(1234.5678, If(1<0,[1, 2, 3]))
+Blank()
+
+>> Round([12.34, 56.78], If(1<0,1))
+[12,57]
+
+>> Round(If(1<0,2), [1, 2, 3])
+Table({Result:0},{Result:0},{Result:0})
+
+// *********** 2+-argument functions, empty tables ***********
+
+>> Concatenate(Filter(["hello", "hi"], Len(Value)>10), " world")
+Table()
+
+>> Concatenate("hello ", Filter(["world", "day"],Len(Value)>10))
+Table()
+
+>> Find(Filter(["hello", "hi"], Len(Value)>10), " world")
+Table()
+
+>> Find("a", Filter(["world", "day"],Len(Value)>10))
+Table()
+
+>> Round(Filter([12.34, 56.78], Value > 100), 1)
+[]
+
+>> Round(1234.5678, Filter([1, 2, 3], Value > 10))
+Table()
+
+>> Concatenate(Filter(["a","b"], Len(Value)>10), If(1<0,["abcdef", "ghijkl"]))
+Blank()
+
+>> Find(If(1<0,["a","b"]), Filter(["abcdef", "ghijkl"], Len(Value)>10))
+Blank()
```

Comment:
Here are a few cases that I don't think are handled properly yet:

1: If any of the arguments is an error table, then the result must be an error table:

    >> Concatenate(["a","b"], If(1/0<2,["abcdef", "ghijkl"]))
    #Error(Kind=Div0)

    >> Find(If(1/0<2,["a","b"]), ["abcdef", "ghijkl"])
    #Error(Kind=Div0)

2: If any of the arguments is a blank table, then the result must be a blank table:

    >> Concatenate(["a","b"], If(1<0,["abcdef", "ghijkl"]))
    Blank()

    >> Find(If(1<0,["a","b"]), ["abcdef", "ghijkl"])
    Blank()

3: If there are both blank tables and error tables, error take precedence (result is an error). I think this may be working fine:

    >> Concatenate(If(1<0,["a","b"]), If(1/0<2,["abcdef", "ghijkl"]))
    #Error(Kind=Div0)

Reply:
1&3 was working fine, updated for case 2, where even when any arg is blank table. the result is blank.